dapp/src/utils/hooks.js

```diff
@@ -1,57 +1,103 @@
 import { useEffect, useState, useRef } from 'react'
 import { useWeb3React } from '@web3-react/core'
-
-import { injected, connectorsByName, getConnectorImage } from './connectors'
+import { SafeAppConnector } from '@gnosis.pm/safe-apps-web3-react'
+
+import {
+  injected,
+  connectorsByName,
+  getConnector,
+  getConnectorImage,
+} from './connectors'
 import AccountStore from 'stores/AccountStore'
 import analytics from 'utils/analytics'
 
 export function useEagerConnect() {
   const { activate, active } = useWeb3React()
 
-  const [tried, setTried] = useState(false)
+  const [triedInjected, setTriedInjected] = useState(false)
+  const [triedSafeMultisig, setTriedSafeMultisig] = useState(false)
+  const [safeMultisigConnector, setSafeMultisigConnector] = useState(null)
+  const [connector, setConnector] = useState(null)
+  const [connectorIcon, setConnectorIcon] = useState(null)
+
+  // Instantiate Gnosis Safe web3-react connector
+  useEffect(() => {
+    // Requires global.window to be available to must be instantiated inside
+    // useEffect for Next.js to work
+    setSafeMultisigConnector(new SafeAppConnector())
+  }, []) // Do this on mount
+
+  // Attempt to use Gnosis Safe Multisig if available
+  useEffect(() => {
+    async function attemptSafeConnection() {
+      // Safe multisig connector not yet initialised, wait for it
+      if (!safeMultisigConnector) return
+
+      // OK to use Gnosis Safe?
+      const canUseGnosisSafe =
+        safeMultisigConnector && (await safeMultisigConnector.isSafeApp())
```

Comment:
nit: `safeMultisigConnector` should always result to a valid object here so just: 
```
const canUseGnosisSafe = await safeMultisigConnector.isSafeApp()
```

Reply:
Done